test/server/test-run-test.js

```diff
@@ -0,0 +1,248 @@
+var expect          = require('chai').expect;
+var Promise         = require('pinkie');
+var noop            = require('lodash').noop;
+var TestRun         = require('../../lib/test-run');
+var commands        = require('../../lib/test-run/commands');
+var CLIENT_MESSAGES = require('../../lib/test-run/client-messages');
+
+function wait (ms) {
+    return new Promise(function (resolve) {
+        setTimeout(resolve, ms);
+    });
+}
+
+function nextTick () {
+    return wait(0);
+}
+
+describe('TestRun', function () {
+    it('Should send the pending command to the client and resolve/reject it when it is done', function () {
+        var testMock = {
+            fixture: 'fixture',
+            fn:      noop
+        };
+
+        var testRun          = new TestRun(testMock);
+        var click1           = commands.createClickCommand();
+        var click2           = commands.createClickCommand();
+        var commandsToClient = [];
+        var testRunDone      = false;
+
+        testRun.once('done', function () {
+            testRunDone = true;
+        });
+
+        var doneCommandResult = {
+            status: 'done'
+        };
+
+        var actionError         = 'action-error';
+        var failedCommandResult = {
+            status: 'failed',
+            err:    actionError
+        };
+
+        function sendClientRequest (commandResult) {
+            return testRun[CLIENT_MESSAGES.ready](commandResult)
+                .then(function (command) {
+                    commandsToClient.push(command);
+                });
+        }
+
+        sendClientRequest()
+            .then(function () {
+                // NOTE: simulate page reload before pending step is
+                // executed that leads to repeated ready message.
```

Comment:
// NOTE: simulates page reload before the pending step is
// executed. This leads to a repeating ready message.